opencensus-migration/src/main/java/OpenTelemetryAdapter.java

```diff
@@ -0,0 +1,9 @@
+import io.opencensus.trace.unsafe.CtxUtils;
+
+public class OpenTelemetryAdapter {
+  private OpenTelemetryAdapter() {}
+
+  public static void register() {
+    CtxUtils.setContextManager(new OpenTelemetryContextManager());
```

Comment:
One potential confusion that we need to think about here: at the moment end users will need a few steps to enable "OpenTelemetry" mode in OpenCensus Java:
1. add a dependency on `opencensus-migration`
2. (implicit) register `OpenTelemetryTraceComponent` - will happen automatically via code in `opencensus-java`
3. (explicit) register context manager (this line)

So `2.` will happen as soon as they add a dependency, and users have no way of disabling that (other than deleting dependency). But they may omit the step `3.` by mistake.

IIUC, what will happen in this situation is all spans will be exported to OpenTelemetry anyway, but the parent-child relationship will be incorrect in some scenarios.

I guess even if that happens it's not a catastrophe, and we just need to provide a clear documentation that both steps are required for correct behavior.

Alternatively, we may think about a solution for that problem.
e.g. perform logic in `OpenTelemetryAdapter.register()` implicitly when dependency is present (probably via the same reflection mechanism as `2.` used for TraceComponent - try to detect if the `OpenTelemetryContextManager ` class exists and instantiate it, or fall back to the default context manager.

Does that make sense?

Reply:
I've added the reflection here: https://github.com/census-instrumentation/opencensus-java/pull/2059/files#diff-0fc325a05eef09a58565e484afffa856465879e2f23590262568a54a239cc6abR33-R52
Let me know if this doesn't match what you're expecting!

Reply:
Awesome! LGTM, then we don't need this file anymore AFAIU.